### DISCUSSION TITLE
réseau Social Mastodon

### DISCUSSION THREAD

**Discussion by linuxmrcmd** [AmauriC/tarteaucitron.js#1121] [Ideas]
Bonjour, 

Apres maintes recherche, j'ai enfin trouvé le widget permettant de récupérer les pouets (posts) du réseau social et de les intégrer au niveau sité.
ce Widget sera mis dans ce site https://www.guppyland.org/file/mastodon/mastodon.widget.js
/```
/ mastodon
tarteaucitron.services.mastodon = {
    "key": "mastodon",
    "type": "social",
    "name": "Mastodon",
    "uri": "https://mastodon.org/privacy-policy",
    "needConsent": true,
    "cookies": [],
    "js": function () {
        "use strict";

		tarteaucitron.fallback(['tacmastodon'], '');
        tarteaucitron.addScript('https://www.guppyland.org/file/mastodon/mastodon.widget.js)';
    },
    "fallback": function () {
        "use strict";
        var id = 'mastodon';
        tarteaucitron.fallback(['tacMastodon'], tarteaucitron.engage(id));
    }

};

**Comment by linuxmrcmd** [comment]


**Reply by AmauriC** [reply]
Il ne me faut pas un zip, juste le marqueur html souvent une balise <script> ou <iframe>

**Reply by linuxmrcmd** [reply]
le nom du script est mastodon.widget.js

**Reply by linuxmrcmd** [reply]
Quel marqueur, le marqueur html souvent une balise <script> ou <iframe>?
Sans précision
J'ai l'impression d'être un imbecile qui ne comprends ce que tu veux

**Reply by AmauriC** [reply]
Il ne faut pas le prendre comme ça 🥺

Par exemple pour Twitter:
```
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s); js.id=id; js.src=p+'://platform.twitter.com/widgets.js'; fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script>
```

Pour Youtube:
```
<iframe width="width" height="height" src="//www.youtube.com/embed/video_id" frameborder="0" allowfullscreen></iframe>
```

Il me faut quelque chose de similaire pour le "convertir" en version tarteaucitron.

**Reply by linuxmrcmd** [reply]
La connexion se fait via une Api, c'est dans le fichier que j'ai transmis
De plus, pour Mastodon, tu crées une application avec id secret et id_account
le second jd est ici
<script>
$(document).ready(function() {
// jQUERY is required!
  var mapi = new MastodonApi({
    target_selector  : '#myTimeline'
    ,instance_uri    : 'https://mastodon.social'
    ,access_token    : 'code secret'
    ,account_id      : 'id'
    // optional parameters
    // ===================
    // - status max
    //,toots_limit     : 5
    // - if you are using font-awesome:
    //,pic_icon        : '<i class="fa fa-picture-o"></i>'
    // or a picture
    //,pic_icon        : '<img src="mypicicon.gif" />'
  });
});
</script>
<div class="mastodon-timeline mastodon-timeline-dark" id="myTimeline"></div>

c'est comme cela qui permet les post dans le site

**Reply by AmauriC** [reply]
Ok je pense comprendre, Mastodon ne propose pas directement un système de widget c'est un service tiers qui récupere les infos via l'api? (ex: https://github.com/YalePrivacyLab/mastodon-timeline-widget/blob/master/docs/widget/README_en.md)

Si oui ce ne sera pas compatible avec tarteaucitron : /

**Reply by linuxmrcmd** [reply]
OUi, c'est cela
D'ailleurs ill faut un code secret généré lors de la creation de l'application.
Je n'ai pas testé ce lien que tu indiques
Ne pas oublier que en plus, les instances sont choisies par l'utilisateur, donc changement possible et en avoir plusieurs.
d'où notre échange

**Reply by linuxmrcmd** [reply]
Je ne vois pas de différence avec un Widget et une api?
Quel est le niveau d'incompatibilité?
merci d'apporter des précisions

**Reply by AmauriC** [reply]
Parce qu'il y a l'air de n'avoir que des paramètres custom. Le script principal n'est pas chargé depuis Mastodon mais depuis le site lui même? Qu'est ce que tu utilises comme plugin pour l'intégrer?

**Reply by AmauriC** [reply]
Le fichier n'existe pas encore donc ça ne m'aide pas. 
D'où viens ce widget? Il doit bien y avoir une doc quelque part?

**Reply by AmauriC** [reply]
Merci c'est plus clair. 
Il existe des dizaines de librairies différentes (https://docs.joinmastodon.org/client/libraries/) et ça ne fait que des requêtes vers un API, ce n'est pas le type de chose que tarteaucitron gère ou qui doivent obtenir le consentement de l'utilisateur, ce serait trop compliqué de prendre en charge tout ça : / désolé

**Reply by linuxmrcmd** [reply]
L'api que j'utilise est https://nomdelinstance/api/v1/accounts/
